.github/scripts/license_years.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2024 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+import re
+from git import Repo
+
+# Initialize a git Repo object
+repo = Repo('.')
+git = repo.git
+
+# Get the last common ancestor of the current branch and the staging branch
+current_branch = repo.active_branch
+common_ancestor_hash = repo.git.merge_base('staging', current_branch.name)
+
+# Check if the current branch hash is the same as the staging hash
+if current_branch.commit.hexsha == common_ancestor_hash:
+  print("There are no commits between the current branch and the staging branch")
+  exit()
+
+# Get the commits that are in the current branch but not in main
+all_commits = repo.iter_commits()
+commits = []
+for commit in all_commits:
+  if commit.hexsha == common_ancestor_hash:
+    break
+  commits.append(commit.hexsha)
+
+# Check if the year in the license header is correct for each file in each commit
+all_valid = True
+for commit in commits:
+  obj = repo.commit(commit)
+  commit_year = obj.committed_datetime.year
+  if len(obj.parents) == 0:
+    # Must be an initial commit
+    continue
+  parent = obj.parents[0]
+  # get the diff of the commit
+  diffs = parent.diff(obj)
+  for diff in diffs:
+    # Only check added or modified files
+    if diff.change_type in ['A', 'M']:
```

Comment:
Factor this whole thing out into a function `get_changed_files(commit)`

Reply:
Also, the source code of [`iter_change_type`](https://github.com/gitpython-developers/GitPython/blob/f4ce70974d8ca0e4d2699b4d5f1016880815ae39/git/diff.py#L222) suggest that you need more than just look at `change_type`. To be safe, change this to:

```python
def iter_change_types(diff_index, change_types):
    for change_type in change_types:
        yield from diff_index.iter_change_type(change_type)
```